.github/workflows/blt-test.yml

```diff
@@ -0,0 +1,85 @@
+name: Validation & tests via BLT.
+
+on:
+  workflow_call:
+    inputs:
+      container:
+        description: 'The docker image to use'
+        type: string
+        required: true
+      source_key:
+        description: 'The source cache key suffix, or empty to checkout'
+        type: string
+        default: 'composer'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  validate:
+    name: 'Validate'
+    runs-on: ubuntu-latest
+    container: ${{ inputs.container }}
+    steps:
+      - name: Restore build from a previous step.
+        uses: actions/cache@v3
+        id: restore-build
+        with:
+          path: ./*
+          # We are only interested in a direct hit.
+          key: build-${{ github.sha }}-${{ inputs.source_key }}
+
+      - name: Abort if we have a source build cache miss.
+        run: exit 1
+        if: steps.restore-build.outputs.cache-hit != 'true'
+
+      - name: Disable telemetry.
+        run: ./vendor/bin/blt blt:telemetry:disable --no-interaction
+
+      - name: Run BLT validate.
+        run: ./vendor/bin/blt validate --no-interaction
+
+  test:
+    name: 'Test'
+    runs-on: ubuntu-latest
+    container: ${{ inputs.container }}
+    services:
+      db:
+        image: mysql:5.7
+        ports: [ 3306 ]
+        env:
+          MYSQL_USER: db
+          MYSQL_PASSWORD: db
+          MYSQL_DATABASE: db
+          MYSQL_ROOT_PASSWORD: root
+        options: --health-cmd="mysqladmin ping" --health-interval=5s --health-timeout=2s --health-retries=3
+    steps:
+      - name: Restore build from a previous step.
+        uses: actions/cache@v3
+        id: restore-build
+        with:
+          path: ./*
+          # We are only interested in a direct hit.
+          key: build-${{ github.sha }}-composer
+
+      - name: Abort if we have a source build cache miss.
+        run: exit 1
+        if: steps.restore-build.outputs.cache-hit != 'true'
+
+      - name: Configure Drupal settings.
+        run: |
+          cp docroot/sites/default/settings.ci.php docroot/sites/default/settings.local.php
+          ./vendor/bin/drush status
+
+      - name: Run BLT test.
+        env:
+          SIMPLETEST_BASE_URL: 'http://127.0.0.1:8888'
+        # drush runserver requires a full bootstrap, so use php -S directly.
+        run: |
+          (cd docroot && php -S 127.0.0.1:8888 .ht.router.php > /tmp/php-log.txt 2>&1 & )
+          ./vendor/bin/phpunit -c phpunit.xml docroot/modules/custom/
```

Comment:
Ah, nice work figuring this out 👏 I could NOT get tests running when doing GHA work earlier this year. 

If someone wanted/needed to override the default implementation (eg there's a custom phpunit file somewhere else, or they wanted to only scan a subset of files) is it easy to add a way to customise this command? Or should people in those situations just copy this file into their repo and edit the command?

(my previous job had a very particular setup, just thinking outside the box - not something we need to implement I don't think)